rclcpp/include/rclcpp/node_interfaces/node_base.hpp

```diff
@@ -15,27 +15,42 @@
 #ifndef RCLCPP__NODE_INTERFACES__NODE_BASE_HPP_
 #define RCLCPP__NODE_INTERFACES__NODE_BASE_HPP_
 
+#include <atomic>
+#include <functional>
 #include <memory>
+#include <mutex>
 #include <string>
+#include <unordered_map>
 #include <vector>
 
 #include "rcl/node.h"
+#include "rclcpp/callback_group.hpp"
 #include "rclcpp/context.hpp"
 #include "rclcpp/macros.hpp"
 #include "rclcpp/node_interfaces/node_base_interface.hpp"
 #include "rclcpp/visibility_control.hpp"
 
+
 namespace rclcpp
 {
 namespace node_interfaces
 {
 
+class map_of_mutexes;
+
 /// Implementation of the NodeBase part of the Node API.
 class NodeBase : public NodeBaseInterface
 {
 public:
   RCLCPP_SMART_PTR_ALIASES_ONLY(NodeBase)
 
+  static std::unique_ptr<map_of_mutexes> map_object_ptr;
```

Comment:
why an unique_ptr?
Can't this just be an `static map_of_mutexes map_object;`

Reply:
Will do

Reply:
I think that was because ```this->map_object = map_of_mutexes()``` was not allowed as the class member ```std::mutex internal_mutex``` is copyable. To make the map_of_mutex static member ptr to an object, I'll have to change the ```internal_mutex``` as a pointer so that the copy constructor is valid.

Reply:
> I think that was because in `node_base` constructor, `this->map_object = map_of_mutexes()`

But we shouldn't need that at all.  Once we have `map_of_mutexes` in static storage, there is no need to do an assignment to another static within the class; just use the global that already exists.